modules/table/src/main/java/org/apache/ignite/internal/table/TableImpl.java

```diff
@@ -141,6 +143,11 @@ public InternalTable internalTable() {
         return tbl;
     }
 
+    @Override
+    public <T extends Partition> PartitionManager<T> partitionManager() {
+        return (PartitionManager<T>) new HashPartitionManagerImpl(tbl, schemaReg, marshallers);
```

Comment:
How is it going to work with `RangePartition`?

Reply:
It is going to be a breaking change, right? So we do some future-proofing with `Partition` interface, but not here?

Reply:
As discussed in private, we don't need to generify `Table`, and this method will have a `Class<T>` parameter to validate the partition type.